src/dynamic_bridge.cpp

```diff
@@ -554,8 +554,8 @@ int main(int argc, char * argv[])
         return;
       }
 
-      ros1_publishers.clear();
-      ros1_subscribers.clear();
+      std::map<std::string, std::string> current_ros1_publishers;
+      std::map<std::string, std::string> current_ros1_subscribers;
```

Comment:
should we use "active_ros1_publishers"/"active_ros1_subscribers" to match the naming of the equivalent "active_ros1_services" variable? In general the naming doesn't match between publishers/subscribers and services variables but I think we can at least do this bit in this PR.